File: PocketBeagle-2/button/python/main.py
"""
This example demonstrates reading GPIO Buttons using the GPIO Keys kernel driver
"""

from chardev import CharDev

BUTTONS_NAME = "buttons"
RIGHT_CODE = 106


btn = CharDev.input_device_by_name(BUTTONS_NAME)

print("Waiting for Input")
while True:
    evt = btn.read_evt()

    # We use value to only print on pressed (not released)
    if evt.code == RIGHT_CODE and evt.value == 1:
        print("Right")

File: PocketBeagle-2/button/rust/src/main.rs
//! This example demonstrates reading button inputs using GPIO Keys driver.

use beagle_helper::chardev::CharDev;
use std::fs::OpenOptions;

const BUTTONS_NAME: &str = "buttons";

fn main() {
    let mut evt =
        CharDev::open_input_with_name(BUTTONS_NAME, OpenOptions::new().read(true)).unwrap();

    loop {
        let inp = evt.read_evt().unwrap();
        match (inp.code, inp.value) {
            // We use value to only print on pressed (not released)
            (106, 1) => println!("Right"),
            _ => {}
        }
    }
}
